apps/dashboard/src/components/template-store/workflow-sidebar.tsx

```diff
@@ -0,0 +1,193 @@
+import { motion } from 'motion/react';
+import { ReactNode } from 'react';
+import { Badge } from '../primitives/badge';
+import { WorkflowMode } from './types';
+
+interface WorkflowSidebarProps {
+  selectedCategory: string;
+  onCategorySelect: (category: string) => void;
+  mode: WorkflowMode;
+}
+
+interface SidebarButtonProps {
+  icon: ReactNode;
+  label: string;
+  onClick?: () => void;
+  isActive?: boolean;
+  bgColor?: string;
+  asChild?: boolean;
+  hasExternalLink?: boolean;
+  beta?: boolean;
+}
+
+const buttonVariants = {
+  initial: { scale: 1 },
+  hover: { scale: 1.01 },
+  tap: { scale: 0.99 },
+};
+
+const iconVariants = {
+  initial: { rotate: 0 },
+  hover: { rotate: 5 },
+};
+
+function SidebarButton({
+  icon,
+  label,
+  onClick,
+  isActive,
+  bgColor = 'bg-blue-50',
+  asChild,
+  beta,
+  hasExternalLink,
+}: SidebarButtonProps) {
+  const ButtonWrapper = asChild ? CreateWorkflowButton : motion.button;
+  const content = (
+    <div className="flex items-center gap-3">
+      <motion.div variants={iconVariants} className={`rounded-lg p-[5px] ${bgColor}`}>
+        {icon}
+      </motion.div>
+      <span className="text-label-sm text-strong-950">{label}</span>
+      {hasExternalLink && (
+        <motion.div whileHover={{ x: 2 }} transition={{ type: 'spring', stiffness: 300 }} className="ml-auto">
+          <ExternalLink className="text-foreground-600 h-3 w-3" />
+        </motion.div>
+      )}
+    </div>
+  );
+
+  return (
+    <ButtonWrapper
+      variants={buttonVariants}
+      initial="initial"
+      whileHover="hover"
+      whileTap="tap"
+      type="button"
+      onClick={onClick}
+      className={`flex w-full items-center gap-2 rounded-xl border border-transparent p-1.5 transition-colors hover:cursor-pointer hover:bg-gray-100 ${
+        isActive ? '!border-[#EEEFF1] bg-white' : ''
+      }`}
+    >
+      {asChild ? (
+        content
+      ) : (
+        <div className="flex w-full items-center gap-2">
+          {content}{' '}
+          {beta && (
+            <Badge color="gray" size="sm">
+              BETA
+            </Badge>
+          )}
+        </div>
+      )}
+    </ButtonWrapper>
+  );
+}
+
+const useCases = [
+  {
+    id: 'popular',
+    icon: <LayoutGrid className="h-3 w-3 text-gray-700" />,
+    label: 'Popular',
+    bgColor: 'bg-blue-50',
+  },
+  {
+    id: 'events',
+    icon: <Calendar className="h-3 w-3 text-gray-700" />,
+    label: 'Events',
+    bgColor: 'bg-blue-50',
+  },
+  {
+    id: 'authentication',
+    icon: <KeyRound className="h-3 w-3 text-gray-700" />,
+    label: 'Authentication',
+    bgColor: 'bg-green-50',
+  },
+  {
+    id: 'social',
+    icon: <Users className="h-3 w-3 text-gray-700" />,
+    label: 'Social',
+    bgColor: 'bg-purple-50',
+  },
+] as const;
+
+const createOptions = [
+  {
+    icon: <FileText className="h-3 w-3 text-gray-700" />,
+    label: 'Blank workflow',
+    bgColor: 'bg-green-50',
+    asChild: true,
+  },
```

Comment:
I don't like this one as it creates this nested modals hell. We have a blank as a separate option under the create button.

![Screenshot 2025-01-14 at 17 31 24](https://github.com/user-attachments/assets/01f3427d-0530-4d48-baa2-c9462571e19a)

Reply:
Will discuss this with Naveen to consider another option in a next iteration. cc. @twentyone24